homeassistant/components/airq/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""The air-Q integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from aioairq import AirQ
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, MANUFACTURER, TARGET_ROUTE, UPDATE_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+class AirQCoordinator(DataUpdateCoordinator):
+    """Coordinator is responsible for querying the device at a specified route."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialise a custom coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=UPDATE_INTERVAL),
+        )
+        session = async_get_clientsession(hass)
+        self.airq = AirQ(
+            entry.data[CONF_IP_ADDRESS], entry.data[CONF_PASSWORD], session
+        )
+        self.device_id = entry.unique_id
+        assert self.device_id is not None
+        self.device_info = DeviceInfo(
+            manufacturer=MANUFACTURER,
+            identifiers={(DOMAIN, self.device_id)},
+        )
+        self.device_info.update(entry.data["device_info"])
+
+    async def _async_update_data(self) -> dict:
+        """Fetch the data from the device."""
+        data = await self.airq.get(TARGET_ROUTE)
```

Comment:
Future improvement: We probably want to handle at least invalid authentication here and raise `ConfigEntryAuthFailed` and add a reauth step in the config flow.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#reauthentication

The parent coordinator class will handle aiohttp errors by default but if there are other library specific errors raised here we should catch them and raise `UpdateFailed` if it's not bad credentials.